File: dormitory-service-go/internal/model/entity/camera.go
package entity

import (
	"time"

	"github.com/sims/campusvision/dormitory-service-go/internal/model/jsontype"
)

// DormCamera maps to the dorm_camera table.
// This is the camera information table storing RTSP stream configuration and status.
type DormCamera struct {
	ID              int64                `db:"id" json:"id"`
	CameraID        string               `db:"camera_id" json:"camera_id"`
	Name            string               `db:"name" json:"name"`
	Building        string               `db:"building" json:"building"`
	RtspURL         string               `db:"rtsp_url" json:"rtsp_url"`
	Direction       string               `db:"direction" json:"direction"`
	Resolution      string               `db:"resolution" json:"resolution"`
	Status          string               `db:"status" json:"status"`
	FPSCurrent      jsontype.NullFloat64 `db:"fps_current" json:"fps_current"`
	TotalFrames     jsontype.NullInt64   `db:"total_frames" json:"total_frames"`
	LastHeartbeat   jsontype.NullTime    `db:"last_heartbeat" json:"last_heartbeat"`
	LastEventTime   jsontype.NullTime    `db:"last_event_time" json:"last_event_time"`
	Enabled         bool                 `db:"enabled" json:"enabled"`
	ConfigJSON      jsontype.NullString  `db:"config_json" json:"config_json"`
	Remark          jsontype.NullString  `db:"remark" json:"remark"`
	LastHealthCheck jsontype.NullTime    `db:"last_health_check" json:"last_health_check"`
	PasswordEnc     jsontype.NullString  `db:"password_enc" json:"password_enc"`
	Nonce           jsontype.NullString  `db:"nonce" json:"nonce"`
	Type            string               `db:"type" json:"type"`
	Protocol        string               `db:"protocol" json:"protocol"`
	Host            jsontype.NullString  `db:"host" json:"host"`
	Port            jsontype.NullInt64   `db:"port" json:"port"`
	Path            jsontype.NullString  `db:"path" json:"path"`
	Username        jsontype.NullString  `db:"username" json:"username"`
	KeyID           jsontype.NullString  `db:"key_id" json:"key_id"`
	CreatedAt       time.Time            `db:"created_at" json:"created_at"`
	UpdatedAt       time.Time            `db:"updated_at" json:"updated_at"`
}

File: frontend/src/api/camera.ts
import request from '@/utils/request';

/** 摄像头实体 */
export interface Camera {
  id: number;
  camera_id: string;
  name: string;
  building: string;
  rtsp_url: string;
  direction: string;
  status: string;
  fps_current: number | null;
  total_frames: number | null;
  last_heartbeat: string | null;
  last_event_time: string | null;
  last_health_check: string | null;
  enabled: boolean;
  config_json: string | null;
  remark: string | null;
  created_at: string;
  updated_at: string;
}

/** 查询摄像头列表 */
export function listCameras(building?: string) {
  return request({ url: '/sims/dorm/cameras', method: 'get', params: { building } }).then((res: any) => {
    if (res?.data?.length > 0) {
      console.log('=== listCameras last item rtsp_url ===', res.data[res.data.length - 1]?.rtsp_url);
    }
    return res;
  });
}

/** 查询单个摄像头详情 */
export function getCamera(id: string) {
  return request({ url: `/sims/dorm/cameras/${id}`, method: 'get' });
}

/** 新增摄像头 */
export function addCamera(data: Partial<Camera>) {
  console.log('=== addCamera request data ===', JSON.stringify(data));
  return request({ url: '/sims/dorm/cameras', method: 'post', data });
}

/** 修改摄像头 */
export function updateCamera(id: string, data: Partial<Camera>) {
  return request({ url: `/sims/dorm/cameras/${id}`, method: 'put', data });
}

/** 删除摄像头 */
export function deleteCamera(id: string) {
  return request({ url: `/sims/dorm/cameras/${id}`, method: 'delete' });
}

/** 查询摄像头状态 */
export function getCameraStatus(id: string) {
  return request({ url: `/sims/dorm/cameras/${id}/status`, method: 'get' });
}

/** 触发健康检查 */
export function healthCheck(id: string) {
  return request({ url: `/sims/dorm/cameras/${id}/health-check`, method: 'post' });
}

/** 查询摄像头快照列表 */
export function getCameraSnapshots(id: string, page: number, size: number) {
  return request({
    url: `/sims/dorm/cameras/${id}/snapshots`,
    method: 'get',
    params: { page, size },
  });
}

/** 查询网关摄像头健康状态（stream-gateway 管理端口） */
export function getGatewayCameraHealth(cameraId: string) {
  return request({
    url: `/api/gateway/cameras/${cameraId}/health`,
    method: 'get',
    headers: {
      'X-Management-Key': import.meta.env.VITE_GATEWAY_MGMT_KEY || '',
      isToken: false,
    },
    baseURL: '',
  });
}
